ocaml/runtime/domain.c

```diff
@@ -1784,11 +1784,11 @@ static void handover_finalisers(caml_domain_state* domain_state)
 
   if (f->todo_head != NULL || f->first.size != 0 || f->last.size != 0) {
     /* have some final structures */
-    if (caml_gc_phase != Phase_sweep_and_mark_main) {
```

Comment:
I think this surprises me. I'd expect something that used to be ok during marking to still be ok during marking.

Reply:
Yeah, that seems fine. The important thing here is that we're not in one of the ephemeron processing subphases.